FasterXML/jackson-module-scala
# shade jackson-module-scala for version 2.9.10-fs0

There are a handful of steps involved here:

 0. add our shaded core jackson fat jar and shaded test jackson jar
    (required only for running the tests here) to the repo's top-
    level lib directory (they are unmanaged dependencies in sbt's
    eyes)
 1. update build configuration to ban upstream jackson jars
 2. update core jackson imports to use their shaded variants
 3. shade jackson-module-scala package namespace
 4. update some hard-coded package name strings in tests
 5. run tests
 6. bump the version in `version.sbt` and publish the shaded jar. For
    whatever asinine reason, publishLocal here builds the javadoc,
    sources, and pom correctly but constructs a compiled jar bereft
    of actual class files. My workaround for this was to publishLocal
    first and then package
 7. test using the shaded jar locally
 8. upload the shaded jar, javadoc, sources, and pom to artifactory.
    if we are still on scala 2.11, upload the 2.12 jars as well.
 9. make a new branch on our github fork off the upstream release tag
    (call it something like 2.9-fs), open a pr against that for
    review, then merge your changes and push a new git tag for the
    patch release (eg. jackson-module-scala-2.9.10-fs0)

Code-wise, this is a dead simple namespace refactor. Unfortunately,
most of the complexity involved comes at runtime in the form of
manually validating the classpath is properly constructed. The repo's
test suite certainly helps and is a good place to start, however some
other checks to make:

 - in sbt, run `show test:fullClasspath` and verify it shows no
   versions of jackson other than our shaded fat jar
 - after packaging the shaded jar, unzip it and verify it contains
   only shaded jackson-module-scala class files
 - as an added sanity check, always unzip our shaded core jackson fat
   jar and make sure it doesn't include unshaded class files either

Assuming you have done your due diligence on all of the above, your
newly shaded jar should be good to go!
